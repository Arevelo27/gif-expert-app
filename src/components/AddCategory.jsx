import PropTypes from 'prop-types';
import { useState } from 'react'

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const onResetForm = () => {
        setInputValue('');
    }

    const onSubmit = (e) => {
        e.preventDefault();
        const newValue = inputValue.trim();
        if (newValue.length <= 1) return;

        onNewCategory(newValue);
        onResetForm();
    }

    return (
        <form onSubmit={onSubmit}>
            <div>
                <input
                    type="text"
                    placeholder="Search gifs"
                    value={inputValue}
                    onChange={onInputChange} />
            </div>
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}
