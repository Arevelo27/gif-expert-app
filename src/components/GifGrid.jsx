import { getGifs } from "../helpers/getGifs";
import { useEffect, useState } from "react";
import { GifItem } from "./GifItem";

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([]);

    const getImages = async (category) => {
        const newImages = await getGifs(category);
        setImages(newImages);
    }

    useEffect(() => {
        getImages(category)
    }, [category])

    return (
        <>
            <h3>{category}</h3>
            <div className="card-grid">
                {
                    images.map((image) => (
                        <GifItem
                            key={image.id}
                            {...image}
                        />
                    ))
                }
            </div>

        </>
    )
}