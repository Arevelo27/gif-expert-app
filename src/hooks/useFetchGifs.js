import { useEffect, useState } from 'react';
import { getGifs } from '../helpers/getGifs';


export const useFetchGifs = ( category ) => {

    const [images, setImages] = useState([]);
    const [isLoading, setIsLoading] = useState(true); // Agregar estado para isLoading

    const getImages = async () => {
        const newImages = await getGifs(category);
        setImages(newImages);
        setIsLoading(false); // Cambiar el estado de isLoading a false despuÃ©s de cargar las imÃ¡genes
    }

    useEffect(() => {
        getImages()
    }, [])

    return {images, isLoading}; 
}