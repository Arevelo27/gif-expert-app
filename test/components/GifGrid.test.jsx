import { render, screen } from '@testing-library/react';
import { GifGrid } from '../../src/components/GifGrid';
import { useFetchGifs } from '../../src/hooks/useFetchGifs';

jest.mock('../../src/hooks/useFetchGifs')

describe('Prueba en <GifGrid/>', () => {

    const category = 'One Punch';

    test('debe hacer match con el snapshot', () => {

        useFetchGifs.mockReturnValue([[], false]); // Devuelve un array

        const { container } = render(<GifGrid category={category} />);
        expect(container).toMatchSnapshot();

    });

    test('debe de mostrar el loading inicialmente', () => {

        useFetchGifs.mockReturnValue([[], true]); // Devuelve un array

        render(<GifGrid category={category} />);
        expect(screen.getByText('Cargando...')).toBeTruthy();

    });

    test('debe de mostrar items cuando se cargan las imagenes useFetchGifs', () => {

        const gifs = [
            { id: '1', title: 'Gif 1', url: 'https://example.com/gif1.gif' },
            { id: '2', title: 'Gif 2', url: 'https://example.com/gif2.gif' }
        ];

        useFetchGifs.mockReturnValue([gifs, false]); // Devuelve un array
        render(<GifGrid category={category} />);
        expect(screen.getByText(category)).toBeTruthy();
        expect(screen.getAllByRole('img').length).toBe(gifs.length);

    });


})
