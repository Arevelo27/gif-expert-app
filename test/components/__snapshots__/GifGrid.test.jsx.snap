// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifGrid/> debe hacer match con el snapshot 1`] = `
<div>
  <h3>
    One Punch
  </h3>
  <div
    class="card-grid"
  />
</div>
`;
